### ISSUE TITLE
Question: What version of HTML does vscode intend to support?

### ISSUE THREAD

**Issue by Tyriar** [Microsoft/vscode#806]
I've opened a few PRs related to HTML support/completion. Just wondering what the official stance is on what version of HTML is supported, currently I've been working against [W3C HTML5.1 WD](http://www.w3.org/TR/html51/) and excluding deprecated tags as their use is discouraged.

**Comment by aeschli** [comment]
We wanted to avoid to clutter the list with deprecated, experimental and not-yet standard tags and attributes. So the process has been to check https://developer.mozilla.org/en-US/docs/Web/HTML/Element on how many browsers already support the feature and add it when there are at least 2 browsers. Deprecated features are removed.
What do you think?
Adding @dstorey as well for an opinion.

**Comment by Tyriar** [comment]
Sounds reasonable.

**Comment by aeschli** [comment]
Closing for now. Please comment if you disagree on the suggested guideline.

**Comment by dstorey** [comment]
TL;DR 2 browser rule of thumb is good, but have room to be flexible if one browser plus strong interest from other browsers. Need to decide if two browsers with same engine count as 2 or 1.

I would somewhat agree with this, although MDN isn't always as up to date as it could be for browser support; especially for MS Edge.

I agree to no deprecated things, or non-standard elements/attributes. 

I would probably expand to include things that are either in development or high priority. If VS Code is always going to be updated every month then there is less of an issue (we just need to remember to update it) as the updates will get out fast enough. Each browser has a dashboard of what is planned/in development now. The only thing to watch out for is that WebKit's doesn't mean it will land in the Safari browser.

A question is, due Chrome and Opera, or IE and Edge count as two browsers, or one?

The nice thing about the CSS Intellisense is one can add new things while editing the file, and it will hide them until there is browser support, so you're less likely to forget about features. This may be less of a problem for HTML, where there is one spec with a good index of elements/attributes.

Some examples:

* keygen. This is well supported (except IE/Edge), but it has been removed from the WhatWG version of the spec, and Chrome want to remove it (and Mozilla are discussing it). As there is so little usage in the wild of keygen, and there is common agreement between browser vendors that it is a bad API, then removing keygen from VSCode makes sense to me, even though it currently has 2+ implementations.

* math (mathML in HTML). I didn't include this as more niche usecase than SVG, IE/Edge has no plans and Chrome removed support for it. It is currently included in 2 browsers though. Also, like SVG this would only really be useful if full support is added rather than just the root element mentioned in the spec.

* menu/menuitem. I included this as I was including standard elements with one implementation. We currently have this set as a lower priority so may make sense to remove it for now, but I do see the accessibility benefit of this element.

* dialog. Also really only has one implementation (but two browsers due to Opera). I'd leave this one though as it is more likely we’ll implement this and it has fairly high public support: https://wpdev.uservoice.com/forums/257854-microsoft-edge-developer/suggestions/6508895-dialog-element

* SMIL. We don't cover SVG yet, so this isn't a problem for now, but Edge/IE won't implement (favouring modern ways of animating with Web Animations, Motion Path, CSS animations, etc) and Chrome are planning to deprecate it. I'd edge towards not including it, unless Chrome change their mind. 

Perhaps a harder question is which attributes to add, as WHATWG and W3C diverge in a number of places. There are less thorough docs on what are supported too. I just added all of them whether they were in WhatWG or W3C. Attributes filter down better (not many custom attributes per element) so the clutter is less of an issue.

**Comment by aeschli** [comment]
@dstorey Thanks a lot for the comment and the interesting insights.
I would say we treat IE and Edge as one browser, as well as Chrome and Opera. Ok for you? Is Safari very different from Chrome? So in the end it would just be about Chrome/Opera/Safari, IE/Edge and Firefox. 
I'm also fine top be flexible and make exceptions if there are good arguments that user will profit.

**Comment by Tyriar** [comment]
WebKit and Blink have diverged a decent amount at this point, so I don't think Safari should be lumped with Chromium.

**Comment by dstorey** [comment]
for the early stuff where it was the same implementation it makes sense to combine, but most/all of those stuff are already documented,so yeah, counting safari/chrome as different makes sense to me.
